emcc.py

```diff
@@ -2320,6 +2322,23 @@ def do_binaryen(final, target, asm_target, options, memfile, wasm_binary_target,
           passes.append('minifyWhitespace')
         final = shared.Building.js_optimizer_no_asmjs(final, passes)
       if DEBUG: save_intermediate('postclean', 'js')
+  # replace placeholder strings with correct subresource locations
+  if shared.Settings.SINGLE_FILE:
+    f = open(final, 'r')
+    js = f.read()
+    f.close()
+    f = open(final, 'w')
+    for tup in [
+      (wasm_text_target, shared.FilenameReplacementStrings.WASM_TEXT_FILE),
+      (wasm_binary_target, shared.FilenameReplacementStrings.WASM_BINARY_FILE),
+      (asm_target, shared.FilenameReplacementStrings.ASMJS_CODE_FILE)
```

Comment:
Hm, I'm mobile and can't review the code right this second, but will the mem init file not always have already been inserted by this point?